extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/QuarkusErrorHandler.java

```diff
@@ -306,23 +379,41 @@ private ContentTypes() {
         private static final String APPLICATION_JSON = "application/json";
         private static final String TEXT_JSON = "text/json";
         private static final String TEXT_HTML = "text/html";
+        private static final String TEXT_PLAIN = "text/plain";
         private static final String APPLICATION_XHTML = "application/xhtml+xml";
         private static final String APPLICATION_XML = "application/xml";
         private static final String TEXT_XML = "text/xml";
 
         // WARNING: The order matters for wildcards: if text/json is before text/html, then text/* will match text/json.
-        private static final Collection<MIMEHeader> SUPPORTED = Arrays.asList(
+        private static final MIMEHeader[] BASE_HEADERS = {
                 new ParsableMIMEValue(APPLICATION_JSON).forceParse(),
                 new ParsableMIMEValue(TEXT_JSON).forceParse(),
                 new ParsableMIMEValue(TEXT_HTML).forceParse(),
                 new ParsableMIMEValue(APPLICATION_XHTML).forceParse(),
                 new ParsableMIMEValue(APPLICATION_XML).forceParse(),
-                new ParsableMIMEValue(TEXT_XML).forceParse());
+                new ParsableMIMEValue(TEXT_XML).forceParse()
+        };
+
+        private static final Collection<MIMEHeader> SUPPORTED = new ArrayList<>(Arrays.asList(BASE_HEADERS));
+        private static final Collection<MIMEHeader> SUPPORTED_CURL = new ArrayList<>();
+        static {
+            SUPPORTED_CURL.add(new ParsableMIMEValue(TEXT_PLAIN).forceParse());
+            SUPPORTED_CURL.addAll(Arrays.asList(BASE_HEADERS));
+            ((ArrayList<MIMEHeader>) SUPPORTED).add(new ParsableMIMEValue(TEXT_PLAIN).forceParse());
+        }
 
         static String pickFirstSupportedAndAcceptedContentType(RoutingContext context) {
             List<MIMEHeader> acceptableTypes = context.parsedHeaders().accept();
-            MIMEHeader result = context.parsedHeaders().findBestUserAcceptedIn(acceptableTypes, SUPPORTED);
-            return result == null ? null : result.value();
+
+            String userAgent = context.request().getHeader("User-Agent");
+            if (userAgent != null && (userAgent.toLowerCase(Locale.ROOT).startsWith("wget/")
+                    || userAgent.toLowerCase(Locale.ROOT).startsWith("curl/"))) {
```

Comment:
At some point we should make this a little nice so we can easily add more values. 
I am thinking of the following HTTP clients that I personally use:

`httpie`, `IntelliJ HTTP Client` and `Postman` (although I am not sure what values they use for the header - if any)